src/api/dirmonitor/polling.c

```diff
@@ -0,0 +1,126 @@
+#include <stdlib.h>
+#include <string.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/stat.h>
+#include <time.h>
+#include <SDL.h>
+
+#define EVENT_MODIFIED (1 << 0)
+
+struct file_info {
+  int fd;
+  time_t mtime;
```

Comment:
Do we want to support higher resolution timestamps when available? This is probably supported by system.get_file_info().

Reply:
we could, `struct stat` exposes the time as a `struct timespec`, which includes nano seconds, my concern here was that `mtim` was only introduced with `POSIX.1-2008` while `mtime` was available since, at least, `POSIX.1-2001`

Reply:
also I double checked, `get_file_info` also makes use of `mtime`

Reply:
Maybe an optional fallback? It does seem like a pretty good thing to do, considering how we tried to fix this for autoreload as well.

Reply:
> also I double checked, `get_file_info` also makes use of `mtime`

There was two PRs to try to get rid of that #1826 is the one I remember.

Reply:
hmm, the code looks reasonable but I am not entirely certain of the use of doubles since they generally come with a performance cost that systems that would rely on this backend can't spare.

I'll probably just use a `struct timespec` for both and only fill out the available space, but I'll have to see

Reply:
The double is purely for Lua. I believe you are comparing this in C, so there shouldn't be any problems comparing the structs directly.